src/gui/SampleWaveform.cpp

```diff
@@ -28,64 +28,39 @@ namespace lmms::gui {
 
 void SampleWaveform::visualize(Parameters parameters, QPainter& painter, const QRect& rect)
 {
-	const int x = rect.x();
-	const int height = rect.height();
-	const int width = rect.width();
-	const int centerY = rect.center().y();
+	if (!parameters.buffer || parameters.size == 0) { return; }
 
-	const int halfHeight = height / 2;
+	const auto centerY = rect.center().y();
+	const auto halfHeight = rect.height() / 2;
```

Comment:
```suggestion
	const int centerY = rect.center().y();
	const int halfHeight = rect.height() / 2;
```
I like this better, you may disagree. It makes it easier to check for subtraction from unsigned types.

Reply:
Check my reply in Discord for why I use it. There's a lot of background into why some choose to do this, but I won't necessarily dislike it if someone chose not to do it in their own PRs.